broker-core/src/main/java/io/zeebe/broker/workflow/processor/gateway/ExclusiveSplitHandler.java

```diff
@@ -44,7 +43,7 @@ public void handle(BpmnStepContext<ExecutableExclusiveGateway> context) {
 
       if (sequenceFlow != null) {
         value.setElementId(sequenceFlow.getId());
-        value.setBpmnElementType(BpmnElementType.SEQUENCE_FLOW);
+        value.setBpmnElementType(sequenceFlow.getElementType());
```

Comment:
Instead of setting the element type and element id in each handler, we could change the method `appendNewEvent()` to 
`context.getOutput().appendNewEvent(WorkflowInstanceIntent.SEQUENCE_FLOW_TAKEN, value, sequenceFlow);` So we can not forget to set the element type.